eval/eval.go

```diff
@@ -671,9 +671,9 @@ func evalMapInfixExpression(operator token.Type, left, right object.Object) obje
 	rightMap := right.(object.Map)
 	switch operator { //nolint:exhaustive // we have default.
 	case token.EQ:
-		return object.MapEquals(leftMap, rightMap)
+		return object.NativeBoolToBooleanObject(object.MapEquals(leftMap, rightMap))
```

Comment:
like here @ccoVeille

Reply:
I had already marked the file as read, so when I saw the refactoring I thought it was no longer used

Reply:
np and btw thanks a lot for all your time reviewing all these PRs